tools/Raven.Migrator/MigrationHelpers.cs

```diff
@@ -0,0 +1,168 @@
+
+                var isFirstDocument = new Reference<bool> { Value = true };
+                foreach (var collection in configuration.CollectionsToMigrate)
+                {
+                    var mongoCollectionName = collection.Key;
+                    var ravenCollectionName = collection.Value;
+                    if (string.IsNullOrWhiteSpace(ravenCollectionName))
+                        ravenCollectionName = mongoCollectionName;
+
+                    await migrateSingleCollection(mongoCollectionName, ravenCollectionName, isFirstDocument, streamWriter);
+                }
+
+                if (migrateAttachments != null)
+                    await migrateAttachments(isFirstDocument, streamWriter);
+
+                await streamWriter.WriteAsync("]}");
+            }
+        }
+
+        public static async Task WriteDocument(
+            ExpandoObject document,
+            string documentId,
+            string collectionName,
+            List<string> propertiesToRemove,
+            Reference<bool> isFirstDocument,
+            StreamWriter streamWriter,
+            List<Dictionary<string, object>> attachments = null)
+        {
+            var dictionary = (IDictionary<string, object>)document;
+            if (propertiesToRemove != null)
+            {
+                foreach (var toRemove in propertiesToRemove)
+                {
+                    dictionary.Remove(toRemove);
+                }
+            }
+
+            if (attachments == null || attachments.Count == 0)
+            {
+                dictionary["@metadata"] = new Dictionary<string, object>
+                {
+                    {RavenDocumentId, documentId},
+                    {RavenCollection, collectionName}
+                };
+            }
+            else
+            {
+                dictionary["@metadata"] = new Dictionary<string, object>
+                {
+                    {RavenDocumentId, documentId},
+                    {RavenCollection, collectionName},
+                    {"@flags", "HasAttachments"},
+                    {"@attachments", attachments}
+                };
+            }
+
+            if (isFirstDocument.Value == false)
+                await streamWriter.WriteAsync(",");
+            isFirstDocument.Value = false;
+
+            var jsonString = JsonConvert.SerializeObject(document);
+            await streamWriter.WriteAsync(jsonString);
+        }
+
+        public static async Task<Dictionary<string, object>> WriteAttachment( 
+            Stream attachmentStream, long totalSize, string attachmentId, 
+            string collectionName, string contentType, long attachmentNumber,
+            Reference<bool> isFirstDocument, StreamWriter streamWriter)
+        {
+            if (isFirstDocument.Value == false)
+                await streamWriter.WriteAsync(",");
+            isFirstDocument.Value = false;
+
+            // we are going to recalculate the hash when importing
```

Comment:
Sure? 
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs#L571

Reply:
correct. need to calculate it here as well.

Reply:
done